### PR TITLE
#949: return null profile when CAS authentication failed

### PR DESCRIPTION
**Pull request by PhanNN**
#949: return null profile when CAS authentication failed

### CODE REVIEW

File: pac4j-cas/src/main/java/org/pac4j/cas/credentials/authenticator/CasRestAuthenticator.java [hunk 1]
```diff
@@ -49,8 +55,13 @@ public void validate(final UsernamePasswordCredentials credentials, final WebCon
             throw new TechnicalException("Credentials are required");
         }
         final String ticketGrantingTicketId = requestTicketGrantingTicket(credentials.getUsername(), credentials.getPassword(), context);
-        final CasRestProfile profile = new CasRestProfile(ticketGrantingTicketId, credentials.getUsername());
-        credentials.setUserProfile(profile);
+		if (Strings.isNullOrEmpty(ticketGrantingTicketId)) {
+			credentials.setUserProfile(null);
```
Comment (leleuj) [inline pac4j-cas/src/main/java/org/pac4j/cas/credentials/authenticator/CasRestAuthenticator.java]: This is not necessary as it is `null` by default.

File: pac4j-cas/src/main/java/org/pac4j/cas/credentials/authenticator/CasRestAuthenticator.java [hunk 2]
```diff
@@ -49,8 +55,13 @@ public void validate(final UsernamePasswordCredentials credentials, final WebCon
             throw new TechnicalException("Credentials are required");
         }
         final String ticketGrantingTicketId = requestTicketGrantingTicket(credentials.getUsername(), credentials.getPassword(), context);
-        final CasRestProfile profile = new CasRestProfile(ticketGrantingTicketId, credentials.getUsername());
-        credentials.setUserProfile(profile);
+		if (Strings.isNullOrEmpty(ticketGrantingTicketId)) {
```
Comment (leleuj) [inline pac4j-cas/src/main/java/org/pac4j/cas/credentials/authenticator/CasRestAuthenticator.java]: I'd prefer to use the `CommonHelper.assertNotBlank` here.

**Comment by leleuj** [comment]
It looks good. Two comments nonetheless....

File: pac4j-cas/src/main/java/org/pac4j/cas/credentials/authenticator/CasRestAuthenticator.java (hunk 2 above)
Comment (PhanNN) [inline pac4j-cas/src/main/java/org/pac4j/cas/credentials/authenticator/CasRestAuthenticator.java]: org.pac4j.core.exception.TechnicalException: TGT cannot be blank
	at org.pac4j.core.util.CommonHelper.assertTrue(CommonHelper.java:134)
	at org.pac4j.core.util.CommonHelper.assertNotBlank(CommonHelper.java:145)

CommonHelper.assertNotBlank("TGT", ticketGrantingTicketId);
@leleuj: I added this line before set profile. But I think it's not correct. We still receive TechnicalException for CAS failed authentication.
Is it ok?

Comment (leleuj) [inline pac4j-cas/src/main/java/org/pac4j/cas/credentials/authenticator/CasRestAuthenticator.java]: I meant `CommonHelper.isNotBlank()` to test if the string is blank...

**Comment by PhanNN** [comment]
@leleuj ok, maybe it's last change :)

**Comment by leleuj** [comment]
Thanks

**Pull request merged**